src/components/ProductInfo/index.js

```diff
@@ -1,56 +1,96 @@
+import { useState } from "react";
 import PropTypes from "prop-types";
-import { Search, Submit } from "../../styled";
+import { useCart } from "../../hooks";
+import { formatter } from "../../utils";
+import { Flex, Button } from "../../styled";
 import {
   Container,
   Title,
   Price,
   Sku,
   Category,
   Tags,
-  Form,
   Description,
+  Stock,
 } from "./styled";
+import Quantity from "../Quantity";
 
-function ProductInfo({ name, price, sku, category, description, specs, tags }) {
-  const handleSubmit = (event) => {
-    event.preventDefault();
+function ProductInfo(props) {
+  const [quantity, setQuantity] = useState(1);
+  const { dispatch, add, products } = useCart();
+  const { format } = formatter;
+  const { product = {} } = props;
+
+  const currQty = products.find((p) => p.id === product.id)?.quantity || 0;
+  const isDisabled = currQty === product.data.stock;
+
+  const isDownDisabled = quantity === 1;
+  const isUpDisabled = quantity + currQty > props.stock - 1;
+
+  const handleAddToCart = () => {
+    dispatch(add({ product: product, quantity: quantity }));
+    setQuantity(1);
+  };
+
+  const handleIncrement = () => {
+    setQuantity((prevQty) => prevQty + 1);
+  };
+
+  const handleDecrement = () => {
+    setQuantity((prevQty) => prevQty - 1);
   };
 
   return (
     <Container>
-      <Title>{name}</Title>
-      <Price>Price: ${price}</Price>
-      <Sku>SKU: {sku}</Sku>
-      <Category>Category: {category.slug}</Category>
+      <Title>{props.name}</Title>
+      <Price>Price: {format(props.price)}</Price>
+      <Sku>SKU: {props.sku}</Sku>
+      <Stock>Stock: {props.stock}</Stock>
+      <Category>Category: {props.category.slug}</Category>
       <Tags>
         <span>Tags:</span>
         <ul>
-          {tags.map((tag) => (
+          {props.tags.map((tag) => (
             <li key={tag}>{tag}</li>
           ))}
         </ul>
       </Tags>
 
       <Description>
-        {description.length > 0 ? description[0].text : ""}
+        {props.description.length > 0 ? props.description[0].text : ""}
       </Description>
 
-      <Form onSubmit={handleSubmit}>
-        <Search
-          type="number"
-          placeholder="Enter quantity"
-          aria-label="Product Quantity"
-        />
-        <Submit type="submit" value="Add to Cart" />
-      </Form>
+      {props.stock === 0 ? (
+        <h2>Product out of stock</h2>
+      ) : (
+        <Flex
+          container
+          flexDirection="column"
+          alignItems="center"
+          margin="15px 0"
+          gap="10px"
+        >
+          <Quantity
+            quantity={isDisabled ? 0 : quantity}
+            handleIncrement={handleIncrement}
+            handleDecrement={handleDecrement}
+            isDownDisabled={isDownDisabled}
+            isUpDisabled={isUpDisabled}
+          />
+          <Button onClick={handleAddToCart} disabled={isDisabled}>
+            {isDisabled ? "Out of stock" : "Add to Cart"}
+          </Button>
```

Comment:
Since "Out of stock" is just an informative message for the user, I recommend using a different tag than `button`. Otherwise the screen readers would read this as: `BUTTON, disabled, Out of stock`. Which misleads the user to think this is an action item.